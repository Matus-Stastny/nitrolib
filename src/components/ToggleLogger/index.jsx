// @flow strict
import * as React from "react";

import Toggle from "../Toggle";
import Core from "./components/Core";

type Data = {|
  open: boolean,
  onToggle: () => void,
|};

type Props = {|
  onOpen: () => void,
  children: (data: Data) => React.Node,
|};

const ToggleLogger = ({ onOpen, children }: Props) => (
  <Toggle>
    {({ open, onToggle }) => (
      <Core open={open} onToggle={onToggle} onOpen={onOpen}>
        {children}
      </Core>
    )}
  </Toggle>
);

export default ToggleLogger;
