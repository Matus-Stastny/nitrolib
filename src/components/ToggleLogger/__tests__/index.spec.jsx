// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import ToggleLogger from "..";

describe("#ToggleLogger", () => {
  test("render", () => {
    const onOpen = jest.fn();
    const wrapper = shallow(
      <ToggleLogger onOpen={onOpen}>
        {({ open, onToggle }) => (
          <div onToggle={onToggle} open={open}>
            LOL
          </div>
        )}
      </ToggleLogger>,
    );

    expect(wrapper.dive().shallow()).toMatchSnapshot();
  });
});
