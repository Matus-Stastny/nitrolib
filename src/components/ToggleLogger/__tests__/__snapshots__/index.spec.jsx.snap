// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ToggleLogger render 1`] = `
<div
  onToggle={[Function]}
  open={false}
>
  LOL
</div>
`;
